UndertaleModTool/Editors/UndertaleRoomEditor.xaml.cs

```diff
@@ -1571,20 +1544,164 @@ public static void GenerateSpriteCache(UndertaleRoom room)
+            {
+                if (layer.TilesData.TilesX == 0 || layer.TilesData.TilesY == 0)
+                {
+                    MainWindow.ShowError("Tile data size can't be zero.");
+                    return;
+                }
+
+                OpenFileDialog dlg = new()
+                {
+                    DefaultExt = "csv",
+                    Filter = "CSV table|*.csv|All files|*"
+                };
+
+                if (dlg.ShowDialog() == true)
+                {
+                    uint[][] tileDataNew = (uint[][])layer.TilesData.TileData.Clone();
+                    string[] tileDataLines = null;
+                    try
+                    {
+                        tileDataLines = File.ReadAllLines(dlg.FileName);
+                    }
+                    catch (Exception ex)
+                    {
+                        MainWindow.ShowError($"Error while opening file - \"{ex.Message}\".");
+                        return;
+                    }
+
+                    if (tileDataLines?.Length != tileDataNew.Length)
+                    {
+                        MainWindow.ShowError("Error - selected file line count doesn't match tile layer height.");
+                        return;
+                    }
+
+                    for (int i = 0; i < tileDataLines.Length; i++)
+                    {
+                        uint[] dataRow;
+                        try
+                        {
+                            dataRow = tileDataLines[i].Split(';').Select(x => UInt32.Parse(x)).ToArray();
+                        }
+                        catch (Exception ex)
+                        {
+                            MainWindow.ShowError($"Error while parsing line {i + 1} - \"{ex.Message}\".");
+                            return;
+                        }
+
+                        if (dataRow.Length != layer.TilesData.TilesX)
+                        {
+                            MainWindow.ShowError($"Length of line {i + 1} is not equal to the tile data width.");
+                            return;
+                        }
+
+                        tileDataNew[i] = dataRow;
+                    }
+
+                    layer.TilesData.TileData = tileDataNew;
+
+                    MainWindow.ShowMessage("Imported successfully.");
+                }
+            }
+        }
+
+        private void LayerCanvas_DataContextChanged(object sender, DependencyPropertyChangedEventArgs e)
+        {
+            // raised on layers rearrange
+            if (e.NewValue is Layer layer)
+            {
+                LayerCanvas canvas = sender as LayerCanvas;
+
+                canvas.CurrentLayer = layer;
+                ObjElemDict[layer] = canvas;
+            }
+        }
+        private void LayerCanvas_Initialized(object sender, EventArgs e)
+        {
+            LayerCanvas canvas = sender as LayerCanvas;
+
+            if (canvas?.DataContext is Layer layer)
```

Comment:
Make this return early

Reply:
It's small, so IMO it should stay as is.

Reply:
Still looks cleaner IMO

Reply:
Still looks cleaner IMO

Reply:
I've changed my mind, so it will be changed.